glog.go

```diff
@@ -101,7 +107,17 @@ func Flush() {
 // skipLogger constructs a suggared logger with an additional caller skip to
 // ensure we log the correct line number.
 func skipLogger() *zap.SugaredLogger {
-	return zap.L().WithOptions(zap.AddCallerSkip(1)).Sugar()
+	logging.mu.RLock()
+	cached := (logging.logger != nil && logging.zapGlobal != nil && logging.zapGlobal == zap.L())
+	logging.mu.RUnlock()
+
+	if !cached {
+		logging.mu.Lock()
+		defer logging.mu.Unlock()
+		logging.logger = zap.L().WithOptions(zap.AddCallerSkip(1)).Sugar()
+		logging.zapGlobal = zap.L()
+	}
+	return logging.logger
```

Comment:
change to "return cached".

Otherwise, the race detector will complain that logging.logger is accessed in a racy way.